articles/active-directory/develop/msal-logging-dotnet.md

```diff
@@ -23,36 +23,88 @@ ms.custom: aaddev
 
 In MSAL logging is set at application creation using the `.WithLogging` builder modifier. This method takes optional parameters:
 
-- `Level` enables you to decide which level of logging you want. Setting it to Errors will only get errors
+- `IIdentityLogger` The logging implementation MSAL will send its logs too after chacking if logging is enabled.
 - `PiiLoggingEnabled` enables you to log personal and organizational data (PII) if set to true. By default this is set to false, so that your application does not log personal data.
-- `LogCallback` is set to a delegate that does the logging. If `PiiLoggingEnabled` is true, this method will receive messages that can have PII, in which case the `containsPii` flag will be set to true.
-- `DefaultLoggingEnabled` enables the default logging for the platform. By default it's false. If you set it to true it uses Event Tracing in Desktop/UWP applications, NSLog on iOS and logcat on Android.
-
-```csharp
-class Program
-{
-  private static void Log(LogLevel level, string message, bool containsPii)
-  {
-     if (containsPii)
-     {
-        Console.ForegroundColor = ConsoleColor.Red;
-     }
-     Console.WriteLine($"{level} {message}");
-     Console.ResetColor();
-  }
-
-  static void Main(string[] args)
-  {
-    var scopes = new string[] { "User.Read" };
-
-    var application = PublicClientApplicationBuilder.Create("<clientID>")
-                      .WithLogging(Log, LogLevel.Info, true)
-                      .Build();
-
-    AuthenticationResult result = application.AcquireTokenInteractive(scopes)
-                                             .ExecuteAsync().Result;
-  }
-}
+
+
+#### IIdentityLogger Interface
+```CSharp
+    public interface IIdentityLogger
+    {
+        //
+        // Summary:
+        //     Checks to see if logging is enabled at given eventLogLevel.
+        //
+        // Parameters:
+        //   eventLogLevel:
+        //     Log level of a message.
+        bool IsEnabled(EventLogLevel eventLogLevel);
+
+        //
+        // Summary:
+        //     Writes a log entry.
+        //
+        // Parameters:
+        //   entry:
+        //     Defines a structured message to be logged at the provided Microsoft.IdentityModel.Abstractions.LogEntry.EventLogLevel.
+        void Log(LogEntry entry);
+    }
+```
+
+#### IIdentityLogger Implementation
```

Comment:
Can you add an example of an adapter from Microsoft.Extension.Logging's ILogger as well?

Reply:
You might also want to say that partner libraries provide implementations already of this interface for various environments (in particular ASP.NET Core)